File: clojure/prime-factors/prime_factors.clj
(ns prime_factors)

(defn- divisible? [a b]
  (= (rem a b) 0))

(defn of [number]
  (loop [remaining number
         primes []
         divisor  2]
    (cond
      (< remaining 2) primes
      (divisible? remaining divisor) (recur (/ remaining divisor)
                                            (conj primes divisor)
                                            divisor)
      :else (recur remaining
                   primes
                   (inc divisor)))))

File: ruby/prime-factors/prime_factors.rb
class PrimeFactors
  class << self
    def for(number)
      divisors = []
      divisor = 2

      while number > 1
        if number % divisor == 0
          number /= divisor
          divisors << divisor
        else
          divisor += 1
        end
      end

      divisors
    end
  end
end
